e2e/testcases/helm_sync_test.go

```diff
@@ -289,6 +470,86 @@ func TestHelmNamespaceRepo(t *testing.T) {
 	}
 }
 
+// TestHelmNamespaceRepoWithConfigMap verifies RepoSync can pick up values and updates from ConfigMap references.
+// This test will work only with following pre-requisites:
+// Google service account `e2e-test-ar-reader@${GCP_PROJECT}.iam.gserviceaccount.com` is created with `roles/artifactregistry.reader` for accessing images in Artifact Registry.
+func TestHelmConfigMapNamespaceRepo(t *testing.T) {
+	repoSyncNN := nomostest.RepoSyncNN(testNs, configsync.RepoSyncName)
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.RequireGKE(t),
+		ntopts.RepoSyncPermissions(policy.AllAdmin()), // NS reconciler manages a bunch of resources.
+		ntopts.NamespaceRepo(repoSyncNN.Namespace, repoSyncNN.Name))
+	rs := nomostest.RepoSyncObjectV1Beta1FromNonRootRepo(nt, repoSyncNN)
+
+	remoteHelmChart, err := helm.PushHelmChart(nt, privateNSHelmChart, privateNSHelmChartVersion)
+	if err != nil {
+		nt.T.Fatalf("failed to push helm chart: %v", err)
+	}
+
+	configMapFilePath := "../testdata/repo-sync-helm-cm-update/configmap-1.yaml"
+	nt.T.Logf("Apply the ConfigMap with values to the cluster defined in %s", configMapFilePath)
+	nt.MustKubectl("apply", "-f", configMapFilePath)
+
+	nt.T.Log("Update RepoSync to sync from a private Artifact Registry")
+	rs.Spec.SourceType = string(v1beta1.HelmSource)
+	rs.Spec.Helm = &v1beta1.HelmRepoSync{HelmBase: v1beta1.HelmBase{
+		Repo:                   helm.PrivateARHelmRegistry(),
+		Chart:                  remoteHelmChart.ChartName,
+		Auth:                   configsync.AuthGCPServiceAccount,
+		GCPServiceAccountEmail: gsaARReaderEmail(),
+		Version:                privateNSHelmChartVersion,
+		ReleaseName:            "test",
+		ValuesFrom:             []v1beta1.ValuesFrom{{Kind: "ConfigMap", Name: "foo", Key: "foo.yaml"}},
+	}}
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Add(nomostest.StructuredNSPath(repoSyncNN.Namespace, repoSyncNN.Name), rs))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Update RepoSync to sync from a private Helm Chart without cluster scoped resources"))
+	err = nt.WatchForAllSyncs(nomostest.WithRepoSha1Func(helmChartVersion(privateNSHelmChartVersion)), nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{repoSyncNN: remoteHelmChart.ChartName}))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Validate(rs.Spec.Helm.ReleaseName+"-"+remoteHelmChart.ChartName, testNs, &appsv1.Deployment{},
+		testpredicates.HasLabel("labelsTest", "first")); err != nil {
+		nt.T.Error(err)
+	}
+
+	configMapFilePath = "../testdata/repo-sync-helm-cm-update/configmap-2.yaml"
+	nt.T.Logf("Apply the ConfigMap with values to the cluster defined in %s", configMapFilePath)
+	nt.MustKubectl("apply", "-f", configMapFilePath)
+
+	if err = nt.Watcher.WatchObject(kinds.Deployment(), rs.Spec.Helm.ReleaseName+"-"+remoteHelmChart.ChartName, testNs,
+		[]testpredicates.Predicate{testpredicates.HasLabel("labelsTest", "second")},
+		testwatcher.WatchTimeout(7*time.Minute)); err != nil {
```

Comment:
I suggest using a dynamic timeout here, I expect we need a shorter timeout for kind and a longer timeout for autopilot etc

Reply:
done! Now using 2*the default timeout which looks like is dynamic based on whether the cluster is autopilot or not